sdk/iot/pnp/src/az_iot_pnp_client.c

```diff
@@ -53,32 +54,42 @@ AZ_NODISCARD az_result az_iot_pnp_client_init(
 
 AZ_NODISCARD az_result az_iot_pnp_client_get_user_name(
     az_iot_pnp_client const* client,
-    az_span mqtt_user_name,
-    az_span* out_mqtt_user_name)
+    char* mqtt_user_name,
+    size_t mqtt_user_name_size,
+    size_t* out_mqtt_user_name_length)
 {
   AZ_PRECONDITION_NOT_NULL(client);
-  AZ_PRECONDITION_VALID_SPAN(mqtt_user_name, 0, false);
-  AZ_PRECONDITION_NOT_NULL(out_mqtt_user_name);
+  AZ_PRECONDITION_NOT_NULL(mqtt_user_name);
+  AZ_PRECONDITION(mqtt_user_name_size > 0);
 
-  az_span remainder = mqtt_user_name;
-  az_span user_name;
-  AZ_RETURN_IF_FAILED(
-      az_iot_hub_client_user_name_get(&client->_internal.iot_hub_client, remainder, &user_name));
+  size_t written;
```

Comment:
Nit: consider a `written_size` or `written_length` or `bytes_written` (I have another comment on size v length).  I sometimes use `written` as a pointer argument that traverses through.

Reply:
Actually - not a nit, please don't call this `written` in 1st place.  This is just the `original_mqtt_user_name_length` (needs a word smith).  I'd name it that, as written tends to be something incremented over course of function.

Reply:
okay I have it changed to `hub_user_name_written_length`. let me know if you have an issue with that.

Reply:
@danewalton - nit could it just be `hub_user_name_length`?  The written feels unnecessary to my eye at least.

Reply:
yup no problem will do. for user_name and topic

Reply:
updated @jspaith